Libplanet/Net/Transports/NetMQTransport.cs

```diff
@@ -661,52 +658,43 @@ private void DoBroadcast(object sender, NetMQQueueEventArgs<(Address?, Message)>
                     DateTimeOffset.UtcNow,
                     _appProtocolVersion);
 
-                peers.AsParallel().ForAll(peer =>
+                lock (_dealerLock)
```

Comment:
I think this `lock` scope might be too wide?

Reply:
In order to define a lock in inner scope, we should make a lock for each dealer socket.

Reply:
Doesn't `AsParallel.Forall()` block all other threads until finished?

Reply:
It works like `Task.WhenAll()` as I know, each task work asynchronously. https://docs.microsoft.com/ko-kr/dotnet/standard/parallel-programming/how-to-write-a-simple-parallel-foreach-loop

Reply:
If I'm not mistaken, I think `Task.WhenAll()` is an asynchronous call and `AsParallel().ForAll()` is a synchronous call *from* the calling thread. 🤨

Reply:
My bad, seems like a synchronous call is okay here since broadcasting is done in a single thread. 😗